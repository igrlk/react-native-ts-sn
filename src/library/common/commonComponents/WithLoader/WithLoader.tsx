import React from 'react';

import Loader from 'library/common/commonComponents/Loader';

interface WithLoaderProps {
	loading: boolean;
	children: React.ReactNode;
}

export default function WithLoader({ loading, children }: WithLoaderProps): any {
	return loading ? <Loader /> : children;
}
