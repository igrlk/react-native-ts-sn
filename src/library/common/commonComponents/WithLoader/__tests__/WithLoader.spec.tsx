import React from 'react';
import renderer from 'react-test-renderer';

import WithLoader from '../WithLoader';
import { shallow } from 'enzyme';

describe('WithLoader', () => {
	it('match to snapshot', () => {
		const tree = renderer.create(<WithLoader loading={false}>Some content</WithLoader>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render loader if loading', () => {
		const component = shallow(<WithLoader loading>Some content</WithLoader>);
		expect(component.find('Loader').length).toBe(1);
	});
});
