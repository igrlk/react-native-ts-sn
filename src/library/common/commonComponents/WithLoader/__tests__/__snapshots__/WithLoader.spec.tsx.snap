// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLoader match to snapshot 1`] = `"Some content"`;
